### PR TITLE
Rename `pattern` to `design` in mesheryctl

### LINKED ISSUE #10592: [mesheryctl] Rename `pattern` to `design` command
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

Currently, we have `mesheryctl pattern` command to import, list patterns into Meshery. The objective of the issue is to rename `mesheryctl pattern` to `mesheryctl design`. Subcommands present under `pattern` would be the same, only the name would change

### Implementation

N/A. Only name would change

### Acceptance Tests

Existing tests to be migrated to new name to avoid test failures.

### Mockups

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by alphaX86**
**Notes for Reviewers**

This PR fixes #10592 

This PR is continuation of PR #10603 due to sync issues

Checklist:
- [x] Basic name change in all files and test files
- [x] Test design command test files 
	- [x] Unit tests 
	- [x] Test in real time 
		- [x] apply
		- [x] delete
		- [x] list
		- [x] view
		- [x] onboard
		- [x] offboard
- [x] Complete rename of command
- [x] Update Docs
	- [x] Index page
	- [x] URL rename
	- [x] Pages (can be generated via workflow or manual run here)
- [x] Update in GH Actions
	- [x] ID GH Actions which use `pattern`
	- [x] Rename to `design`
	- [x] Ref PR for cross-repo (PR: https://github.com/layer5io/meshery-smp-action/pull/91)
- [x] Fix lint issues

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/design/apply.go [hunk 1]
```diff
@@ -36,25 +36,25 @@ import (
...
-	"caption": "Usage of mesheryctl pattern apply",
+	"caption": "Usage of mesheryctl design apply",
 }
 
 var applyCmd = &cobra.Command{
 	Use:   "apply",
-	Short: "Apply pattern file",
-	Long:  `Apply pattern file will trigger deploy of the pattern file`,
+	Short: "Apply design file",
+	Long:  `Apply design will trigger deploy of the design file`,
```
Comment (Yashsharma1911) [inline mesheryctl/internal/cli/root/design/apply.go:51]: Not related but as we are here, we can make this long description more helpful

Comment (alphaX86) [reply mesheryctl/internal/cli/root/design/apply.go:51]: Noted @Yashsharma1911

**Comment by sayantan1413** [comment]
Hi @alphaX86 if you are not working on this PR can I take it up?

**Comment by alphaX86** [comment]
> Hi @alphaX86 if you are not working on this PR can I take it up?
> 

Hi @sayantan1413 welcome back! Yes, you can proceed to cherry-pick commits from the PR and start working.

**Comment by sayantan1413** [comment]
Hi @alphaX86 I was reviewing the code and just wanted to understand what all are left here. Do I have to fix only the lint checks which are happening because some test packages are still using "pattern" instead of "design".

**Comment by alphaX86** [comment]
If you find any files using `pattern` then do change to `design`

**Comment by l5io** [comment]
🚀 Preview for commit 8f71b59ddcf04baf6cebe82c1e25bfd966e14de2 at: https://66b8c926059d03579e68f56c--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit c2841881c38ebc807dbad2bef3f1400d872dd561 at: https://66c052faac8fdb37d3807a94--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit fd98edc9fc263e2da96c54103ad8e7e6e4836fce at: https://66c0545f9cb5a3415e653e78--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 8d3ca1eb365e53d2d0c97bdc935dccc1fde83307 at: https://66c05ac9f1bcf72f0c0d60b7--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 87309572a417a781ef8659c445e8573c615b2709 at: https://66c05bba9cb5a347bb653d2c--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
Ready for initial review. Lint fixes in progress

**Comment by l5io** [comment]
🚀 Preview for commit 6cba088f4f0fa116ef0bb7d072616706494770e5 at: https://66c2f7639a4e0db289aa7620--meshery-docs-preview.netlify.app

File: mesheryctl/pkg/utils/error.go [hunk 2]
```diff
@@ -277,11 +277,11 @@ func formatError(msg string, cmd cmdType) string {
...
-		return formatUsageDetails(msg, patternUsageURL)
-	case cmdPatternView:
-		return formatUsageDetails(msg, patternViewURL)
-	case cmdPatternExport:
+	case cmdDesign:
+		return formatUsageDetails(msg, designUsageURL)
+	case cmdDesignView:
+		return formatUsageDetails(msg, designViewURL)
+	case cmdDesignExport:
 		return formatUsageDetails(msg, patternExportURL)
```
Comment (lekaf974) [inline mesheryctl/pkg/utils/error.go:285]:
```suggestion
		return formatUsageDetails(msg, designExportURL)
```

File: mesheryctl/internal/cli/root/relationships/relationship.go [hunk 3]
```diff
@@ -113,7 +113,7 @@ func selectRelationshipPrompt(relationship []relationship.RelationshipDefinition
 
 	for _, _rel := range relationship {
 		// here display Kind and EvaluationQuery as relationship name
-		relationshipName := fmt.Sprintf("kind: %s, EvaluationPolicy: %s, SubType: %s", _rel.Kind, _rel.EvaluationQuery, _rel.SubType)
+		relationshipName := fmt.Sprintf("kind: %s, EvaluationPolicy: %s, SubType: %s", _rel.Kind, *_rel.EvaluationQuery, _rel.SubType)
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/relationships/relationship.go:116]: Why did we added this `*` char

Comment (alphaX86) [reply mesheryctl/internal/cli/root/relationships/relationship.go:116]: Not sure how it got added, let me check this up

File: mesheryctl/internal/cli/root/design/error.go [hunk 4]
```diff
@@ -22,73 +22,81 @@ import (
...
-		[]string{"Unable to fetch Pattern"},
+		[]string{"Unable to fetch Design"},
 		[]string{err.Error()},
-		[]string{"Invalid pattern name or ID"},
-		[]string{"Run `mesheryctl pattern view -a` to view all available patterns."})
+		[]string{"Invalid design name or ID"},
+		[]string{"Run `mesheryctl design view -a` to view all available designs."})
 }
 
 func ErrImportPattern(err error) error {
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/error.go:70]:
```suggestion
func ErrImportDesign(err error) error {
```
Should it be updated ?

File: mesheryctl/internal/cli/root/design/error.go [hunk 5]
```diff
@@ -22,73 +22,81 @@ import (
...
-		[]string{fmt.Sprintf("Invalid pattern source type: `%s`", invalidSourceType)},
-		[]string{"Invalid pattern source type due to wrong type/passing"},
-		[]string{"pattern source type (-s) is invalid or not passed."},
+		[]string{fmt.Sprintf("Invalid design source type: `%s`", invalidSourceType)},
+		[]string{"Invalid design source type due to wrong type/passing"},
+		[]string{"design source type (-s) is invalid or not passed."},
 		[]string{"Ensure you pass a valid source type. \nAllowed source types: %s", strings.Join(validSourceTypes, ", ")})
 }
 
 func ErrPatternManifest() error {
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/error.go:86]:
```suggestion
func ErrDesignManifest() error {
```
should it be updated ?

File: mesheryctl/internal/cli/root/design/error.go [hunk 6]
```diff
@@ -22,73 +22,81 @@ import (
...
-	return errors.New(ErrPatternManifestCode, errors.Alert,
+	return errors.New(ErrDesignManifestCode, errors.Alert,
 		[]string{"No file path detected"},
 		[]string{"No manifest file path detected"},
 		[]string{"Manifest path not provided"},
-		[]string{"Provide the path to the pattern manifest. \n\n%v", errPatternMsg})
+		[]string{"Provide the path to the design manifest. \n\n%v", errPatternMsg})
+}
+
+func ErrOnboardPattern() error {
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/error.go:94]:
```suggestion
func ErrOnboardDesign() error {
```
?

File: mesheryctl/internal/cli/root/design/error.go [hunk 7]
```diff
@@ -22,73 +22,81 @@ import (
...
+	return errors.New(ErrOnboardDesignCode, errors.Alert,
+		[]string{"Error Onboarding design"},
+		[]string{"Unable to onboard design due to empty path"},
+		[]string{"File path or design name not provided."},
+		[]string{"Provide a file path/design name. \n\n%v", errOnboardMsg})
 }
 
 func ErrPatternSourceType() error {
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/error.go:102]:
```suggestion
func ErrDesignSourceType() error {
```

File: mesheryctl/internal/cli/root/design/error.go [hunk 8]
```diff
@@ -99,37 +107,29 @@ func ErrPatternSourceType() error {
...
 
-func ErrOnboardPattern() error {
-	return errors.New(ErrOnboardPatternCode, errors.Alert,
-		[]string{"Error Onboarding pattern"},
-		[]string{"Unable to onboard pattern due to empty path"},
-		[]string{"File path or pattern name not provided."},
-		[]string{"Provide a file path/pattern name. \n\n%v", errOnboardMsg})
-}
-
 func ErrPatternFound() error {
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/error.go:110]:
```suggestion
func ErrDesignNotFound() error {
```

File: mesheryctl/internal/cli/root/design/error.go [hunk 9]
```diff
@@ -99,37 +107,29 @@ func ErrPatternSourceType() error {
...
-		[]string{"Provide an pattern name. Use `mesheryctl pattern list`to display list of patternlications"},
+	return errors.New(ErrDesignFoundCode, errors.Alert,
+		[]string{"design not found"},
+		[]string{"No design found with a given name"},
+		[]string{"design name not provided"},
+		[]string{"Provide an design name. Use `mesheryctl design list`to display list of designs"},
 	)
 }
 
 func ErrViewPatternFlag() error {
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/error.go:119]:
```suggestion
func ErrViewDesignFlag() error {
```

File: mesheryctl/internal/cli/root/design/error.go [hunk 10]
```diff
@@ -99,37 +107,29 @@ func ErrPatternSourceType() error {
...
+	return errors.New(ErrDesignFlagCode, errors.Alert,
 		[]string{"Invalid command"},
 		[]string{"Wrong use of command flags"},
-		[]string{"-a/all flag is being used while an pattern is specified"},
-		[]string{"-a/-all cannot be used when [pattern name|id] is specified"})
+		[]string{"-a/all flag is being used while an design is specified"},
+		[]string{"-a/-all cannot be used when [design name|id] is specified"})
 }
 
 func ErrOffboardPattern(err error) error {
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/error.go:127]:
```suggestion
func ErrOffboardDesign(err error) error {
```

File: mesheryctl/pkg/utils/helpers.go [hunk 11]
```diff
@@ -135,9 +135,9 @@ const (
 	cmdFilterDelete             cmdType = "filter delete"
 	cmdFilterList               cmdType = "filter list"
 	cmdFilterView               cmdType = "filter view"
-	cmdPattern                  cmdType = "pattern"
-	cmdPatternView              cmdType = "pattern view"
-	cmdPatternExport            cmdType = "pattern export"
+	cmdDesign                   cmdType = "design"
+	cmdDesignView               cmdType = "design view"
+	cmdDesignExport             cmdType = "pattern export"
```
Comment (lekaf974) [inline mesheryctl/pkg/utils/helpers.go:140]:
```suggestion
	cmdDesignExport             cmdType = "design export"
```

File: mesheryctl/pkg/utils/helpers.go [hunk 12]
```diff
@@ -67,8 +67,8 @@ const (
 	filterDeleteURL                = docsBaseURL + "reference/mesheryctl/filter/delete"
 	filterListURL                  = docsBaseURL + "reference/mesheryctl/filter/list"
 	filterViewURL                  = docsBaseURL + "reference/mesheryctl/filter/view"
-	patternUsageURL                = docsBaseURL + "reference/mesheryctl/pattern"
-	patternViewURL                 = docsBaseURL + "reference/mesheryctl/pattern/view"
+	designUsageURL                 = docsBaseURL + "reference/mesheryctl/design"
+	designViewURL                  = docsBaseURL + "reference/mesheryctl/design/view"
 	patternExportURL               = docsBaseURL + "reference/mesheryctl/pattern/export"
```
Comment (lekaf974) [inline mesheryctl/pkg/utils/helpers.go:72]:
```suggestion
	designExportURL               = docsBaseURL + "reference/mesheryctl/design/export"
```

File: docs/pages/reference/mesheryctl/mesheryctl-design-apply.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
+---
```
Comment (lekaf974) [inline docs/pages/reference/mesheryctl/mesheryctl-design-apply.md:1]: all those mesheryctl-design-*.md files don't need to be created, they will be autogenerated

**Review by lekaf974 — CHANGES_REQUESTED** [review]
Can you address comments please

**Comment by l5io** [comment]
🚀 Preview for commit a917c1c3ab84b633737baa14c5227704dc9ae59b at: https://66f41f8c6e0d7c0a5b321401--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 8b4f122318b5c606f4f6de5b6fd66729e95ee417 at: https://66f421eae56223049a2b9da1--meshery-docs-preview.netlify.app

File: mesheryctl/internal/cli/root/relationships/relationship.go (hunk 3 above)
Comment (alphaX86) [reply mesheryctl/internal/cli/root/relationships/relationship.go]: Fixed via PR sync

**Comment by alphaX86** [comment]
> Can you address comments please

Suggestions applied in the PR. Please review once

File: mesheryctl/internal/cli/root/design/design_test.go [hunk 14]
```diff
@@ -37,97 +37,106 @@ func TestPatterncmd(t *testing.T) {
...
+			Name:             "design view with ID",
 			Args:             []string{"view", "4o7fbc9b-708d-4396-84b8-e2ba37c1adcc"},
-			ExpectedResponse: "pattern.id.view.output.golden",
+			ExpectedResponse: "design.id.view.output.golden",
 			URLs: []utils.MockURL{
 				{
 					Method:       "GET",
-					URL:          testContext.BaseURL + "api/pattern/",
-					Response:     "pattern.id.view.golden",
+					URL:          testContext.BaseURL + "/api/pattern/4o7fbc9b-708d-4396-84b8-e2ba37c1adcc",
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/design_test.go:103]: if we do a delete on that id the test will fail so not sure it is really relevant

File: mesheryctl/internal/cli/root/design/design_test.go [hunk 15]
```diff
@@ -37,97 +37,106 @@ func TestPatterncmd(t *testing.T) {
...
+				},
+				{
+					Method:       "GET",
+					URL:          testContext.BaseURL + "/api/pattern?page_size=10000",
+					Response:     "view.design.api.response.golden",
+					ResponseCode: 200,
+				},
+				{
+					Method:       "GET",
+					URL:          testContext.BaseURL + "/api/pattern?search=4o7fbc9b-708d-4396-84b8-e2ba37c1adcc",
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/design_test.go:115]: same here

File: mesheryctl/internal/cli/root/design/view.go [hunk 16]
```diff
@@ -121,8 +121,8 @@ mesheryctl pattern view [pattern-name | ID]
 			// use the first match from the result when searching by pattern name
 			arr := dat["patterns"].([]interface{})
 			if len(arr) == 0 {
-				utils.Log.Error(ErrPatternNotFound())
-				utils.Log.Info(fmt.Sprintf("pattern with name: %s not found. Enter a valid pattern name or ID \n", pattern))
+				utils.Log.Error(ErrDesignNotFound())
+				utils.Log.Info(fmt.Sprintf("design with name: %s not found. Enter a valid design name or ID \n", pattern))
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/view.go:125]: Looks a duplicate to me, does this not already displayed in `ErrorDesignNotFound` error ?

File: mesheryctl/internal/cli/root/design/view.go [hunk 17]
```diff
@@ -34,17 +34,17 @@ var (
...
-	"caption": "Usage of mesheryctl pattern view",
+	"caption": "Usage of mesheryctl design view",
 }
 
 var viewCmd = &cobra.Command{
-	Use:   "view pattern name",
-	Short: "Display pattern(s)",
-	Long:  `Displays the contents of a specific pattern based on name or id`,
+	Use:   "view design name",
+	Short: "Display design(s)",
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/view.go:42]:
```suggestion
	Short: "Display a design content",
```

**Review by lekaf974 — COMMENTED** [review]
@alphaX86 added new comments

File: mesheryctl/internal/cli/root/design/view.go (hunk 16 above)
Comment (alphaX86) [reply mesheryctl/internal/cli/root/design/view.go:125]: Noted. Removing this.

I just made a bulk rename and movement of commands from `pattern` to `design`. So I didn't notice these.

**Review by lekaf974 — COMMENTED** [review]
LGTM @Jougan-0 Can you have a second eye on it

**Comment by lekaf974** [comment]
@alphaX86 can you update your branch with main

@MUzairS15 @Jougan-0 can you take a look on this, so we can go ahead and merge it ?

**Comment by l5io** [comment]
🚀 Preview for commit 0abc397e8e3226918c754e3165a23b0cb7984b68 at: https://66fd549e537e8d0615f92b44--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
> 🚨 Alert! Git Police! We couldn’t help but notice that one or more of your commits is missing a sign-off. _A what?_ A commit sign-off (your email address).
> 
> To amend the commits in this PR with your signoff using the instructions provided in the DCO check.
> 
> To configure your dev environment to automatically signoff on your commits in the future, see [these instructions](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md?rgh-link-date=2024-10-02T14%3A09%3A56Z#signing-off-on-commits-developer-certificate-of-origin).
> 
> > Be sure to [join the community](https://slack.meshery.io), if you haven't yet and please leave a ⭐ [star on the project](../stargazers) 😄

Issue temporarily solved up.

**Comment by alphaX86** [comment]
> @alphaX86 can you update your branch with main
> 
> @MUzairS15 @Jougan-0 can you take a look on this, so we can go ahead and merge it ?

Done @lekaf974 

@MUzairS15 @Jougan-0 I need you folks for a quick review. This PR is a long standing one

File: mesheryctl/internal/cli/root/design/delete.go [hunk 18]
```diff
@@ -64,9 +64,9 @@ mesheryctl pattern delete [file | URL]
 			err := utils.DeleteConfiguration(mctlCfg.GetBaseMesheryURL(), pattern, "pattern")
 			if err != nil {
 				utils.Log.Error(err)
-				return errors.Wrap(err, utils.PatternError(fmt.Sprintf("failed to delete pattern %s", args[0])))
+				return errors.Wrap(err, utils.DesignError(fmt.Sprintf("failed to delete design %s", args[0])))
```
Comment (Jougan-0) [inline mesheryctl/internal/cli/root/design/delete.go:67]: If you log here and return the error the error will be logged twice. Please remove one for all the instances.//
@alphaX86

Comment (alphaX86) [reply mesheryctl/internal/cli/root/design/delete.go]: Done

**Review by Jougan-0 — APPROVED** [review]

**Comment by Jougan-0** [comment]
@lekaf974 Approved from my end whenever you feel like please go ahead and merge it.

**Comment by l5io** [comment]
🚀 Preview for commit d5017d9490501968b2b40ea19f3e225ec52f8f0c at: https://66fd75de2515e43ae32ba4af--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
Thanks @Jougan-0 

@lekaf974 could you finally review once on your side before we wrap this one up?

**Comment by l5io** [comment]
🚀 Preview for commit 0fbc084a7681c1f69fe0392f0c148e9c69986841 at: https://66fd78632443dc3aa0ce736b--meshery-docs-preview.netlify.app

**Review by leecalcote — DISMISSED** [review]

**Comment by lekaf974** [comment]
> @lekaf974 could you finally review once on your side before we wrap this one up?

Will do and good job

File: mesheryctl/internal/cli/root/design/delete.go [hunk 19]
```diff
@@ -149,13 +149,13 @@ mesheryctl pattern delete [file | URL]
 
 		req, err = utils.NewRequest("DELETE", deployURL, bytes.NewBuffer(patternFileByt))
 		if err != nil {
-			utils.Log.Error(err)
+			// utils.Log.Error(err)
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/delete.go:152]:
```suggestion
```

File: mesheryctl/internal/cli/root/design/delete.go [hunk 20]
```diff
@@ -149,13 +149,13 @@ mesheryctl pattern delete [file | URL]
...
 			return utils.ErrCreatingRequest(err)
 		}
 
 		res, err := utils.MakeRequest(req)
 		if err != nil {
-			utils.Log.Error(err)
+			// utils.Log.Error(err)
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/delete.go:158]:
```suggestion
```

**Review by lekaf974 — CHANGES_REQUESTED** [review]
@alphaX86 can you remove all unnecessary comments

**Comment by l5io** [comment]
🚀 Preview for commit 0f3f0a12e982e13934671e04643eae74c701ae61 at: https://6700793a69b10341ba8ea92f--meshery-docs-preview.netlify.app

File: mesheryctl/internal/cli/root/design/delete.go [hunk 21]
```diff
@@ -63,10 +63,10 @@ mesheryctl pattern delete [file | URL]
 		if isID {
 			err := utils.DeleteConfiguration(mctlCfg.GetBaseMesheryURL(), pattern, "pattern")
 			if err != nil {
-				utils.Log.Error(err)
-				return errors.Wrap(err, utils.PatternError(fmt.Sprintf("failed to delete pattern %s", args[0])))
+				// utils.Log.Error(err)
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/delete.go:66]:
```suggestion
```

File: mesheryctl/internal/cli/root/design/delete.go [hunk 22]
```diff
@@ -77,16 +77,16 @@ mesheryctl pattern delete [file | URL]
 		if !govalidator.IsURL(file) {
 			content, err := os.ReadFile(file)
 			if err != nil {
-				utils.Log.Error(utils.ErrFileRead(errors.New(utils.PatternError(fmt.Sprintf("failed to read file %s. Ensure the filename or URL is valid", file)))))
-				return utils.ErrFileRead(errors.New(utils.PatternError(fmt.Sprintf("failed to read file %s. Ensure the filename or URL is valid", file))))
+				// utils.Log.Error(utils.ErrFileRead(errors.New(utils.DesignError(fmt.Sprintf("failed to read file %s. Ensure the filename or URL is valid", file)))))
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/delete.go:80]:
```suggestion
```

File: mesheryctl/internal/cli/root/design/delete.go [hunk 23]
```diff
@@ -77,16 +77,16 @@ mesheryctl pattern delete [file | URL]
...
+				return utils.ErrFileRead(errors.New(utils.DesignError(fmt.Sprintf("failed to read file %s. Ensure the filename or URL is valid", file))))
 			}
 
 			patternFileByt = content
 		} else {
 			// Else treat it like a URL
 			url, path, err := utils.ParseURLGithub(file)
 			if err != nil {
-				utils.Log.Error(utils.ErrParseGithubFile(err, file))
+				// utils.Log.Error(utils.ErrParseGithubFile(err, file))
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/delete.go:89]:
```suggestion
```

**Comment by alphaX86** [comment]
> @alphaX86 can you remove all unnecessary comments

ACK! Suggestions applied

**Comment by l5io** [comment]
🚀 Preview for commit 4550e49f8873dd5cac4387456c087c863811ac45 at: https://6700fdf88d7bd9c52fa3bff4--meshery-docs-preview.netlify.app

**Review by lekaf974 — APPROVED** [review]
LGTM

**Comment by lekaf974** [comment]
@alphaX86 ok to merge, Thansks  and good job

**Comment by lekaf974** [comment]
Let me know in slack once merge, I need to update some docuementation

**Comment by alphaX86** [comment]
Thanks @lekaf974. And sure will notify you via Slack once done

**Pull request merged**